packages/infra/lib/surescripts-stack.ts

```diff
@@ -0,0 +1,527 @@
+      maxConcurrency: 4,
+    },
+    waitTime: Duration.seconds(0), // No limit
+  },
+  receiveVerificationResponse: {
+    name: "SurescriptsReceiveVerificationResponse",
+    entry: "surescripts-receive-verification-response",
+    lambda: {
+      memory: 1024,
+      timeout: receiveVerificationLambdaTimeout,
+    },
+    queue: {
+      alarmMaxAgeOfOldestMessage: Duration.hours(1),
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(receiveVerificationLambdaTimeout.toSeconds() * 2 + 1),
+      createRetryLambda: false,
+    },
+    eventSource: {
+      batchSize: 1,
+      reportBatchItemFailures: true,
+      maxConcurrency: 4,
+    },
+    waitTime: Duration.seconds(0), // No limit
+  },
+  receiveFlatFileResponse: {
+    name: "SurescriptsReceiveFlatFileResponse",
+    entry: "surescripts-receive-flat-file-response",
+    lambda: {
+      memory: 2048,
+      timeout: receiveFlatFileResponseLambdaTimeout,
+    },
+    queue: {
+      alarmMaxAgeOfOldestMessage: Duration.hours(1),
+      maxMessageCountAlarmThreshold: 15_000,
+      maxReceiveCount: 3,
+      visibilityTimeout: Duration.seconds(receiveFlatFileResponseLambdaTimeout.toSeconds() * 2 + 1),
+      createRetryLambda: false,
+    },
+    eventSource: {
+      batchSize: 1,
+      reportBatchItemFailures: true,
+      maxConcurrency: 4,
+    },
+    waitTime: Duration.seconds(0), // No limit
+  },
+};
+
+function surescriptsEnvironmentVariables({
+  surescripts,
+  allowReplicaAccess = true,
+  allowBundleAccess = false,
+}: {
+  surescripts: EnvConfig["surescripts"];
+  allowReplicaAccess?: boolean;
+  allowBundleAccess?: boolean;
+}): Record<string, string> {
+  return {
+    SURESCRIPTS_SFTP_HOST: surescripts?.surescriptsHost,
+    SURESCRIPTS_SFTP_SENDER_ID: surescripts?.surescriptsSenderId,
+    SURESCRIPTS_SFTP_SENDER_PASSWORD: surescripts?.secrets.SURESCRIPTS_SFTP_SENDER_PASSWORD,
+    SURESCRIPTS_SFTP_RECEIVER_ID: surescripts?.surescriptsReceiverId,
+    SURESCRIPTS_SFTP_PUBLIC_KEY: surescripts?.secrets.SURESCRIPTS_SFTP_PUBLIC_KEY,
+    SURESCRIPTS_SFTP_PRIVATE_KEY: surescripts?.secrets.SURESCRIPTS_SFTP_PRIVATE_KEY,
+    ...(allowReplicaAccess
+      ? { SURESCRIPTS_REPLICA_BUCKET_NAME: surescripts?.surescriptsReplicaBucketName }
+      : {}),
+    ...(allowBundleAccess
+      ? { SURESCRIPTS_BUNDLE_BUCKET_NAME: surescripts?.surescriptsBundleBucketName }
+      : {}),
+  };
+}
+
+interface SurescriptsNestedStackProps extends NestedStackProps {
+  config: EnvConfig;
+  vpc: ec2.IVpc;
+  alarmAction?: SnsAction;
+  lambdaLayers: LambdaLayers;
+  surescriptsReplicaBucket: s3.Bucket;
+  surescriptsBundleBucket: s3.Bucket;
+}
+
+export class SurescriptsNestedStack extends NestedStack {
+  // SFTP commands
+  readonly connectSftpLambda: Lambda;
+  readonly synchronizeSftpLambda: Lambda;
+  // Data pipeline Lambdas
+  readonly sendPatientRequestLambda: Lambda;
+  readonly sendPatientRequestQueue: Queue;
```

Comment:
If we can, let's use the interfaces to declare these. All good to use the actual class for those needed.

This helps so the codebase as a whole can more easily refer to references instead actual instances.

Reply:
I'm a bit unclear on what you mean by this one, can you clarify?

Reply:
CDK provides interfaces for the types. It's usually preferred to work on a higher level type as it allows for less coupling (`IQueue`, `IFunction`).

But we're talking about the props in the stack that creates the resources, so let's disregard this. It would make more sense if we were building utility functions to work w/ lambdas/queues, then if we could declare the params as the interfaces it would be less coupled.